pydis_site/apps/api/models/bot/tag.py

```diff
@@ -8,17 +9,29 @@
 from pydis_site.apps.api.models.utils import ModelReprMixin
 
 
+def is_bool_validator(value: Any) -> None:
+    """Validates if a given value is of type bool."""
+    if not isinstance(value, bool):
+        raise ValidationError(f"This field must be of type bool, not {type(value)}.")
+
+
 def validate_tag_embed_fields(fields):
     """Raises a ValidationError if any of the given embed fields is invalid."""
     field_validators = {
         'name': (MaxLengthValidator(limit_value=256),),
-        'value': (MaxLengthValidator(limit_value=1024),)
+        'value': (MaxLengthValidator(limit_value=1024),),
+        'inline': (is_bool_validator,),
     }
 
+    required_fields = ('name', 'value')
+
     for field in fields:
         if not isinstance(field, Mapping):
             raise ValidationError("Embed fields must be a mapping.")
 
+        if not all(required_field in field for required_field in required_fields):
+            raise ValidationError("Embed fields must contain both a 'name' and a 'value' field.")
```

Comment:
I would prefer if this error message was kept generic if we add something new to `required_fields`, e.g. something like

```suggestion
            raise ValidationError(f"Embed fields must contain the following fields: {', '.join(required_fields)}.")
```

Reply:
Good point.